packages/delisp-core/src/unify.ts

```diff
@@ -211,13 +211,26 @@ export function unify(
     return success(ctx);
   } else if (t1.type === "boolean" && t2.type === "boolean") {
     return success(ctx);
+  } else if (
+    t1.type === "type-variable" &&
+    t1.userSpecified &&
+    t2.type === "type-variable" &&
```

Comment:
Bit of a nit, but why not simply `(t1.type === "type-variable" && t1.userSpecified && t1 === t2)`? Should be the same and this is now the only branch that explicitly returns an error (instead of letting the final `else` do that).

Reply:
True. I did that because there was not `!t1.userSpecified` in the other case for normal variables. It should be equivalent now.

Although I don't feel very comfortable letting the type checker continue. I wanted to stop the flow as soon as possible so there is no chance a future case will break it.